### PR TITLE
test(NODE-4648): pull prepare shell out to script and fix reporter error handling

### PR DESCRIPTION
**Pull request by nbbeeken**
#### What is changing?

Yoinks prepare-shell out of our yml to make it easily runnable when using spawn hosts.

Our prepare script is currently embedded in our evergreen yml. Pulling it out makes running it locally and on spawn hosts much easier.

Additionally, while attempting pulling out this script it was discovered one of our tasks is failing a lot of tests but reporting success ([NODE-4642](https://jira.mongodb.org/browse/NODE-4642)). It turns out errors thrown from mocha reporters are squashed (visually) but still exit the process, but with success. It's a small change to add error handling to our reporter.

While we're adding error handling to the reporter we should remove stdout/err capturing. We don't use it, and it doesn't correctly work on evergreen, and it make xunit.xml huge.

### Double check the following

- [x] Ran `npm run check:lint` script
- [x] Self-review completed using the [steps outlined here](https://github.com/mongodb/node-mongodb-native/blob/HEAD/CONTRIBUTING.md#reviewer-guidelines)
- [x] PR title follows the correct format: `<type>(NODE-xxxx)<!>: <description>`
- [x] Changes are covered by tests
- [x] New TODOs have a related JIRA ticket

### CODE REVIEW

File: .evergreen/generate_evergreen_tasks.js [hunk 1]
```diff
@@ -91,13 +91,13 @@ BASE_TASKS.push({
...
     },
     { func: 'bootstrap kms servers' },
-    {
-      func: 'run tests',
-      vars: {
-        MONGODB_API_VERSION: '1',
-        NO_EXIT: ''
-      }
-    }
+    // TODO(NODE-XXXX): Fix versioned API tests
```
Comment (dariakp) [inline .evergreen/generate_evergreen_tasks.js:94]: NODE-4642?

Comment (nbbeeken) [reply .evergreen/generate_evergreen_tasks.js]: Yes and the PR should now be NODE-4648

**Comment by nbbeeken** [comment]
this patch runs the entire waterfall to double check these changes work correctly on all platforms / setups:
https://spruce.mongodb.com/version/6324af0b32f417656e62b0cf/tasks

**Comment by nbbeeken** [comment]
We're passing everything, the mongosh tests don't use our env setup

**Review by dariakp — APPROVED** [review]

**Pull request merged**